playlistmanager.lua

```diff
@@ -76,7 +76,7 @@ local settings = {
   system = "auto",
 
   --path where you want to save playlists. Do not use shortcuts like ~ or $HOME
-  playlist_savepath = "/home/anon/Documents/",
+  playlist_savepath = mp.find_config_file('playlists'),
```

Comment:
What does this do? I cannot find the docs for it.

For me it resolves into ~/Documents but I'm not sure why.

Reply:
This function does not have an official document, but you can see it in the official Lua script of MPV. In my opinion, its original function is to find configuration files in MPV's configuration directory and return them, but it seems that it can also return folder paths. Maybe your  MPV's configuration directory is ~/Documents? For me it resolves into D:\MPV\portable_config.
![批注 2020-04-24 205955](https://user-images.githubusercontent.com/22742500/80215233-b1754a80-866e-11ea-97c6-00c124f075d8.png)

Reply:
Nevermind my playlistmanager.conf was overriding it back to Documents.

My only concern with this computed value is it cannot be set by the conf file, so I rather this would be a fallback rather than the default. Like I said I think it's fine if default is nil.

I'm not sure if there is any difference with this and using `command_native({"expand-path", "~~home/"})`

Reply:
The problem now is to find a ’default‘ path that can be automatically calculated to be available in win and * UNIX. Because any relative paths cannot be used in conf files, it just accept a String of absolute path，I think this is a good solution. In my opinion, in a sense, fallback and default mean the same thing.
As for the two functions, I don't think they are different. Maybe it's better to use 
`command_native({"expand-path", `"~~home/"})`,
because mp.find_config_file may be an experimental function.

Reply:
I think the `mpv/playlists` is a great default/fallback.

The reason I don't like it being in the settings as the default is because of the conf
https://github.com/jonniek/mpv-playlistmanager/blob/81af99cdf4f5ae762763c5f00eb3d73a8ce48d53/playlistmanager.conf#L43-L44

The conf has all of the default values of the settings and if we use a computed one it would have to be an exception. I think `nil` is good enough if we update the comment to explain it resolves to `mpv/playlists`.

I think for the average user `find_config_file`, `expand-path` or `nil` are all equally unintuitive so the only good UX is a good comment I think.

So I think something like
```
#absolute path where you want to save playlists. Leave as empty to default to mpv/playlists
playlist_savepath=
```

Reply:
I agree with you.
Perhaps you can use shell commands to parse shortcuts. Is it possible? Can MPV receive the return value of shell command?

Reply:
I'm not sure I understand what you mean

Reply:
I mean, you can use the `echo` command to parse shortcuts. 
For example：
`test=io.popen("echo %appdata%/mpv")
print(test:read("*a"))`
Result：
`C:\Users\username\AppData\Roaming/mpv`
or on Linux，`echo ~` or `echo $HOME` should be able to return the absolute path.
But it doesn't seem to make much sense...

Reply:
I'm not sure if it's necessary to expand shortcuts in a sophisticated manner.

This is how mpv-webm expands home dir: https://github.com/ekisu/mpv-webm/blob/91c3cd666d41b78a823d19c0df9b4f406ed796be/build/webm.lua#L170-L188

I think this would make sense if we don't use the mpv dir as the default. But if the user deliberately changes the dir I don't think we need to do anything fancy.